pkg/registry/extensions/deployment/strategy.go

```diff
@@ -77,6 +77,12 @@ func (deploymentStrategy) PrepareForUpdate(ctx api.Context, obj, old runtime.Obj
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Status = oldDeployment.Status
 
+	if newDeployment.Spec.Strategy.Type == extensions.RecreateDeploymentStrategyType {
```

Comment:
This may not be the right place to do this. I think it makes sense to do it for PATCH of the strategy type and another strategy, but not for a PUT that actually sets the RollingUpdate strategy fields.